src/main/kotlin/org/wycliffeassociates/tstudio2rc/TstudioMetadata.kt

```diff
@@ -0,0 +1,100 @@
+package org.wycliffeassociates.tstudio2rc
+
+import com.fasterxml.jackson.core.JsonFactory
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.fasterxml.jackson.module.kotlin.registerKotlinModule
+import org.wycliffeassociates.resourcecontainer.entity.Checking
+import org.wycliffeassociates.resourcecontainer.entity.DublinCore
+import org.wycliffeassociates.resourcecontainer.entity.Language
+import org.wycliffeassociates.resourcecontainer.entity.Manifest
+import org.wycliffeassociates.resourcecontainer.entity.Source
+import org.wycliffeassociates.tstudio2rc.serializable.ProjectManifest
+import org.wycliffeassociates.tstudio2rc.serializable.TargetLanguage
+import java.io.File
+import java.time.LocalDate
+
+internal class TstudioMetadata(path: String) {
+
+    // navigate to project's root directory
+    private val projectDir = File(path)
+        .walk()
+        .firstOrNull {
+            isBookFolder(it.invariantSeparatorsPath)
+        }!!
```

Comment:
if you're just going to !!, why not first {} instead of firstOrNull {}

Reply:
For `!!`, the exception thrown will be `NullPointerException` with the stacktrace at projectDir, instead of `NoSuchElementException` which is less meaningful